File: scheduler-service/src/main/java/pt/isep/tmdei/schedulerservice/model/response/CreateDeliveryResponseDTO.java
package pt.isep.tmdei.schedulerservice.model.response;

import java.util.Date;

import lombok.Data;
import pt.isep.tmdei.schedulerservice.model.data.CoordinatesDTO;

@Data
public class CreateDeliveryResponseDTO {

    private String username;
    private String deliveryId;
    private CoordinatesDTO pickupCoordinates;
    private CoordinatesDTO dropOffCoordinates;
    private Date created;
    private String status;
    private boolean expedited;

}

File: delivery-management/api/model/response/createDeliveryResponseDTO.js
export class CreateDeliveryResponseDTO {
  constructor(delivery) {
    this.username = delivery.account;
    this.deliveryId = delivery._id;
    this.pickupCoordinates = delivery.pickup;
    this.dropOffCoordinates = delivery.dropOff;
    this.created = delivery.created;
    this.status = delivery.status;
    this.expedited = delivery.expedited;
  }
}
